packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1620,6 +1692,12 @@ bool ArbCore::threadBody(ThreadDataStruct& thread_data) {
                       << "\n";
             return false;
         }
+
+        if (machine_to_cache) {
+            // Add to cache now that database changes are committed
+            combined_machine_cache.timedAdd(std::move(machine_to_cache));
+        }
```

Comment:
Flagging this since I'm totally sure if we still want this change with the other change in @PlasmaPower's PR, but looks reasonable either way

Reply:
It shouldn't be necessary, but I also don't think it can hurt anything

Reply:
probably doesn't matter, but I'll leave it